.github/workflows/sphinx.yml

```diff
@@ -0,0 +1,30 @@
+name: Sphinx build
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Build HTML
+      run: |
+        python -m pip install --upgrade pip 
+        pip install -e . 
+        pip install sphinx sphinx_rtd_theme
+        sphinx-build docs docs/_build/html/
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: html-docs
+        path: docs/_build/html/
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
+      if: github.ref == 'refs/heads/main'
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: docs/_build/html
```

Comment:
You don't need to use a gh-pages branch anymore.  You can use GitHub actions and something like
```
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
        with:
          # Upload built website
          path: 'docs/_build/html'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1

```

Reply:
Nice! Thank you for sharing!